crates/router/src/connector/adyen/transformers.rs

```diff
@@ -2040,6 +2052,188 @@ impl From<AdyenNotificationRequestItemWH> for AdyenResponse {
+    fn try_from(item: &types::PayoutsRouterData<F>) -> Result<Self, Self::Error> {
+        let auth_type = AdyenAuthType::try_from(&item.connector_auth_type)?;
+        let payout_method_data = get_payout_card_details(&item.request.payout_method_data).map_or(
+            Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "payout_method_data",
+            }),
+            Ok,
+        )?;
+        Ok(Self {
+            amount: Amount {
+                currency: item.request.destination_currency.to_string(),
+                value: item.request.amount,
+            },
+            merchant_account: auth_type.merchant_account,
+            payment_method: payout_method_data,
+            reference: item.request.payout_id.clone(),
+            shopper_reference: item.merchant_id.clone(),
+        })
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum AdyenPayoutFulfillRequest {
+    Bank(PayoutBankRequest),
+    Card(Box<PayoutCardRequest>),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayoutBankRequest {
+    pub merchant_account: String,
+    pub original_reference: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayoutCardRequest {
+    pub amount: Amount,
+    pub card: PayoutCardDetails,
+    pub billing_address: Option<Address>,
+    pub merchant_account: String,
+    pub reference: String,
+    pub shopper_name: ShopperName,
+    pub nationality: Option<storage_enums::CountryAlpha2>,
+    pub entity_type: Option<storage_enums::EntityType>,
+}
+
+// Payouts fulfill request transform
+impl<F> TryFrom<&types::PayoutsRouterData<F>> for AdyenPayoutFulfillRequest {
+    type Error = Error;
+    fn try_from(item: &types::PayoutsRouterData<F>) -> Result<Self, Self::Error> {
+        let auth_type = AdyenAuthType::try_from(&item.connector_auth_type)?;
+        let payout_type = item.request.payout_type.to_owned();
+
+        let merchant_account = auth_type.merchant_account;
+        match payout_type {
+            storage_enums::PayoutType::Bank => Ok(Self::Bank(PayoutBankRequest {
+                merchant_account,
+                original_reference: item
+                    .request
+                    .connector_payout_id
+                    .clone()
+                    .unwrap_or("".to_string()),
+            })),
+            storage_enums::PayoutType::Card => Ok(Self::Card(Box::new(PayoutCardRequest {
+                amount: Amount {
+                    value: item.request.amount,
+                    currency: item.request.destination_currency.to_string(),
+                },
+                card: get_payout_card_details(&item.request.payout_method_data).map_or(
+                    Err(errors::ConnectorError::MissingRequiredField {
+                        field_name: "payout_method_data",
+                    }),
+                    Ok,
+                )?,
```

Comment:
Add `get_payout_method_data` in connector/utils.rs so that other can reuse it
```suggestion
        let payout_method_data = get_payout_card_details(&item.request.get_payout_method_data()?);
```